### PR TITLE
Full TypeScript Refactor 

### PR DESCRIPTION
**Pull request by Tower450**
amuse toi bien.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,44 +1,10 @@
...
-# node-waf configuration
-.lock-wscript
-
-# Compiled binary addons (http://nodejs.org/api/addons.html)
+# Compiled binary addons (https://nodejs.org/api/addons.html)
 build/Release
 
 # Dependency directories
 node_modules/
 jspm_packages/
```
Comment (vlnk) [inline .gitignore]: what is jspm package?

File: .gitignore [hunk 2]
```diff
@@ -52,7 +18,7 @@ typings/
 *.tgz
 
 # Yarn Integrity file
-.yarn-integrity
+.yarn-integrit
 
-#Visual Studio configuration
-.vscode
```
Comment (vlnk) [inline .gitignore]: you need to hide your config

File: .gitignore [hunk 3]
```diff
@@ -52,7 +18,7 @@ typings/
 *.tgz
 
 # Yarn Integrity file
-.yarn-integrity
+.yarn-integrit
```
Comment (vlnk) [inline .gitignore]: it won't work

File: src/app/dao/entity.ts [hunk 4]
```diff
@@ -2,109 +2,117 @@
  * Module dependencies.
  */
 import * as Sequelize from 'sequelize'
-import { Entity as entityModel } from '../models/entity'
+import { Entity } from '../models/entity'
 import { Request, Response } from '../../../node_modules/@types/express/index'
 
-export const get = (req: Request, res: Response) => {
```
Comment (vlnk) [inline src/app/dao/entity.ts:8]: i suggest you don't use arrow function for methods, you can use it only for short statement (map, forEach, event) https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions

File: src/app/models/entity.ts [hunk 5]
```diff
@@ -1,17 +1,27 @@
...
-  // updatedAt: 'UpdatedAt'
-})
+@Table(
+  {
+    tableName: 'Entity',
+    timestamps: false,
+    paranoid: false
+  }
+)
+export class Entity extends Model<Entity> {
```
Comment (vlnk) [inline src/app/models/entity.ts]: trim your code

File: src/server.ts [hunk 6]
```diff
@@ -56,7 +54,6 @@ export class Server {
   }
 
   private onListening (server: any) {
-    // console.log(server)
     let addr = server.address()
     let bind = typeof addr === 'string'
       ? 'pipe ' + addr
```
Comment (vlnk) [inline src/server.ts]: i suggest a if instead of a too long one-line-if

Comment (vlnk) [reply src/server.ts]: and set the types

**Review by vlnk — CHANGES_REQUESTED** [review]
do fixes suggested from comments

File: .gitignore (hunk 1 above)
Comment (Tower450) [reply .gitignore]: Done.

File: .gitignore (hunk 3 above)
Comment (Tower450) [reply .gitignore]: Done.

File: .gitignore (hunk 2 above)
Comment (Tower450) [reply .gitignore]: Done.

File: src/app/dao/entity.ts (hunk 4 above)
Comment (Tower450) [reply src/app/dao/entity.ts:8]: Done.

File: src/app/models/entity.ts (hunk 5 above)
Comment (Tower450) [reply src/app/models/entity.ts]: Done.

File: src/server.ts (hunk 6 above)
Comment (Tower450) [reply src/server.ts]: Done.

**Review by vlnk — APPROVED** [review]

**Pull request merged**